blockindex/contractstaking/indexer.go

```diff
@@ -171,11 +171,13 @@ func (s *Indexer) DeleteTipBlock(context.Context, *block.Block) error {
 }
 
 func (s *Indexer) commit(handler *contractStakingEventHandler) error {
-	batch, delta := handler.Result()
+	batch, delta, height := handler.Result()
 	if err := s.cache.Merge(delta); err != nil {
 		s.reloadCache()
 		return err
 	}
+	s.cache.PutHeight(height)
```

Comment:
`height` info should be hold by indexer not a cache unless the cache stores historical data

Reply:
agree, height is just `blk.Height()` in indexer, can directly use there, no need to pass down

1. just change to `commit(handler, height)`
2. can revert the change in `func (eh *contractStakingEventHandler) Result()`

Reply:
agree, `height` should be handled by indexer, cache does not need the concept of height

Reply:
fixed